pkg/image/admission/imagepolicy/rules/rules.go

```diff
@@ -0,0 +1,143 @@
+package rules
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	"github.com/openshift/origin/pkg/image/admission/imagepolicy/api"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+type ImagePolicyAttributes struct {
+	Resource      unversioned.GroupResource
+	Name          imageapi.DockerImageReference
+	OriginalName  string
+	Image         *imageapi.Image
+	ExcludedRules sets.String
+}
+
+type RegistryMatcher interface {
+	Matches(name string) bool
+}
+
+type RegistryNameMatcher imageapi.DefaultRegistryFunc
```

Comment:
We don't support that in image streams, so there's no point to that now.
If you're arguing we should fix it, then it needs to be fixed everywhere.
This is actually the canonical source for names because it's used by the
underlying strategy for images.

On Fri, Aug 5, 2016 at 8:49 AM, David Eads <notifications@github.com> wrote:

> In pkg/image/admission/imagepolicy/rules/rules.go
> <https://github.com/openshift/origin/pull/8995#discussion_r73686602>:
>
> > +	imageapi "github.com/openshift/origin/pkg/image/api"
> > +)
> > +
> > +type ImagePolicyAttributes struct {
> > +	Resource      unversioned.GroupResource
> > +	Name          imageapi.DockerImageReference
> > +	OriginalName  string
> > +	Image         *imageapi.Image
> > +	ExcludedRules sets.String
> > +}
> > +
> > +type RegistryMatcher interface {
> > +	Matches(name string) bool
> > +}
> > +
> > +type RegistryNameMatcher imageapi.DefaultRegistryFunc
>
> yeah, this usage needs to get back the list all possible name aliases.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/8995/files/f96f5e56ba70c7c0f24f875621efdf1ef45ce3c0#r73686602>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p4Jpyl9EDi0GpY3OijwQbqO8DCTXks5qczFygaJpZM4IlDlb>
> .
>